src/MiddlewareListener.php

```diff
@@ -0,0 +1,185 @@
+<?php
+/**
+ * @see       https://github.com/zendframework/zend-mvc-middleware for the canonical source repository
+ * @copyright Copyright (c) 2019 Zend Technologies USA Inc. (http://www.zend.com)
+ * @license   https://github.com/zendframework/zend-mvc-middleware/blob/master/LICENSE.md New BSD License
+ */
+
+namespace Zend\Mvc\Middleware;
+
+use Interop\Http\ServerMiddleware\MiddlewareInterface;
+use Psr\Container\ContainerInterface;
+use Psr\Http\Message\ResponseInterface as PsrResponseInterface;
+use Psr\Http\Message\ResponseInterface;
+use Zend\EventManager\AbstractListenerAggregate;
+use Zend\EventManager\EventManagerInterface;
+use Zend\Http\Response;
+use Zend\Mvc\Application;
+use Zend\Mvc\Exception\InvalidMiddlewareException;
+use Zend\Mvc\MvcEvent;
+use Zend\Psr7Bridge\Psr7Response;
+use Zend\Stratigility\MiddlewarePipe;
+
+class MiddlewareListener extends AbstractListenerAggregate
+{
+    /**
+     * Attach listeners to an event manager
+     *
+     * @param  EventManagerInterface $events
```

Comment:
minor: we don't need php-docs for param/return type as the same as in the method declaration